.github/workflows/release.yml

```diff
@@ -14,40 +14,103 @@ env:
-          target: aarch64-apple-ios
+          toolchain: nightly-2024-01-11
+          components: rust-std
+          target: aarch64-apple-ios,aarch64-apple-darwin,aarch64-apple-ios-sim
 
-      - name: Install Rust Toolchain for aarch64-apple-ios-sim
-        uses: RDXWorks-actions/toolchain@master
-        with:
-          toolchain: ${{ env.RUST_TOOLCHAIN }}
-          components: ${{ env.RUST_COMPONENTS }}
-          target: aarch64-apple-ios-sim
+      - name: Compute release tag
+        id: tags
+        run: |
+          set -euo pipefail
+
+          LAST_TAG=`git tag --sort=creatordate | tail -1`
+          NEXT_TAG=$(echo ${LAST_TAG} | awk -F. -v OFS=. 'NF==1{print ++$NF}; NF>1{$NF=sprintf("%0*d", length($NF), ($NF+1)); print}')
+
+          echo "NEXT_TAG=$NEXT_TAG" >> "$GITHUB_OUTPUT"
+          echo "LAST_TAG=$LAST_TAG" >> "$GITHUB_OUTPUT"
+
+          echo "🚢  NEXT_TAG: $NEXT_TAG"
+          echo "🚢  LAST_TAG: $LAST_TAG"
+
+      - name: Build artifacts
+        id: build
+        run: |
+          set -euo pipefail
+
+          rustup target add aarch64-apple-darwin aarch64-apple-ios aarch64-apple-ios-sim
+
+          echo "🚢 Switch 'useLocalFramework' to 'false' in Package.swift for release"
+          sed -i '' 's/let useLocalFramework = true/let useLocalFramework = false/' Package.swift
+
+          # output is: "<CHKSUM>;<$XCFRAME_ZIP_PATH>"
+          OUTPUT_OF_BUILD=$(bash scripts/ios/build-sargon.sh --release-tag "$NEXT_TAG" | tail -n 1) || exit $?
+          if [[ "$OUTPUT_OF_BUILD" == "BUILT_WITHOUT_RELEASE" ]]; then
+              echo "Error, failed to build, did you forget to pass '--release' to build script? Otherwise check if build-sargon.sh has recently been changed (to something incorrect...)"
+              exit 1;
+          fi
+          CHECKSUM=`echo "$OUTPUT_OF_BUILD" | cut -d ";" -f 1` || exit $?
+          XCFRAME_ZIP_PATH=`echo "$OUTPUT_OF_BUILD" | cut -d ";" -f 2` || exit $?
 
-      - name: Run iOS build script
+          echo "🚢  CHECKSUM: $CHECKSUM"
+          echo "🚢  XCFRAME_ZIP_PATH: $XCFRAME_ZIP_PATH"
+
+          echo "🚢 Ensuring Sargon build for release - that it will work for e.g. iOS wallet to archive."
+          sed -i '' 's/let useLocalFramework = false/let useLocalFramework = true/' Package.swift
+          swift build -c release || exit $?
+          echo "🚢 Swift Sargon builds for release ✅"
+
+          echo "XCFRAME_ZIP_PATH=$XCFRAME_ZIP_PATH" >> "$GITHUB_OUTPUT"
+        env:
+          NEXT_TAG: ${{ steps.tags.outputs.NEXT_TAG }}
+
+      - name: Push release tag
+        if: github.ref == 'refs/heads/main'
+        run: |
+          set -euo pipefail
+
+          # We have .gitigored Sargon.swift because we dont need it in git history, but we
+          # need it for this release, so we must FORCE add it (since it is ignored).
+          echo "🚢  Staging changed files"
+          git add --force Package.swift apple/Sources/UniFFI/Sargon.swift
```

Comment:
@muzuke are you certain that the git state between jobs are shared? I thought jobs ran in parallell? Here we run `git add` which will only work if this job (lacks ID btw...?) `Push release tag` "shares git/file state" with the job `build`.